src/pages/ServiceDetails/ServiceInfo/ServiceInfoVirtualServices/VirtualServiceRoute.tsx

```diff
@@ -0,0 +1,352 @@
+    let checks = checkForPath(
+      validation,
+      'spec/' +
+        this.props.kind.toLowerCase() +
+        '[' +
+        routeIndex +
+        ']/route[' +
+        destinationIndex +
+        ']/weight/' +
+        routeItem.weight
+    );
+    checks.push(
+      ...checkForPath(
+        validation,
+        'spec/' + this.props.kind.toLowerCase() + '[' + routeIndex + ']/route[' + destinationIndex + ']/destination'
+      )
+    );
+
+    let severity = highestSeverity(checks);
+    let iconName = severity ? severityToIconName(severity) : 'ok';
+    if (iconName !== 'ok') {
+      return (
+        <OverlayTrigger
+          placement={'right'}
+          overlay={this.infotipContent(checks)}
+          trigger={['hover', 'focus']}
+          rootClose={false}
+        >
+          <Icon type="pf" name={iconName} />
+        </OverlayTrigger>
+      );
+    } else {
+      return '';
+    }
+  }
+
+  destinationFrom(destinationWeight: DestinationWeight, i: number) {
+    let destination = destinationWeight.destination;
+    return {
+      host: destination.host || '-',
+      subset: destination.subset || '-',
+      port: destination.port ? destination.port.number || '-' : '-'
+    };
+  }
+
+  infotipContent(checks: ObjectCheck[]) {
+    return (
+      <Popover id={this.props.name + '-weight-tooltip'}>
+        {checks.map((check, index) => {
+          return this.objectCheckToHtml(check, index);
+        })}
+      </Popover>
+    );
+  }
+
+  objectCheckToHtml(object: ObjectCheck, i: number) {
+    return (
+      <Row key={'objectCheck-' + i}>
+        <Col xs={1}>
+          <Icon type="pf" name={severityToIconName(object.severity)} />
+        </Col>
+        <Col xs={10} style={{ marginLeft: '-20px' }}>
+          {object.message}
+        </Col>
+      </Row>
+    );
+  }
+
+  bulletChartValues(routes: TCPRoute | HTTPRoute) {
+    return (routes.route || []).map((destinationWeight, u) => ({
+      value: routes.route && routes.route.length === 1 ? 100 : destinationWeight.weight,
+      title: `${u}_${destinationWeight.weight}`,
+      color: PFBlueColors[u % PFBlueColors.length],
+      tooltipFunction: () => {
+        const badges = this.renderDestination(destinationWeight.destination);
+        return (
+          <Tooltip id={`${u}_${destinationWeight.weight}`} key={`${u}_${destinationWeight.weight}`}>
+            <div className="label-collection">{badges}</div>
+          </Tooltip>
+        );
+      }
+    }));
+  }
+
+  renderDestination(destination: Destination) {
+    return (
+      <ul style={{ listStyleType: 'none', paddingLeft: '15px' }}>
+        <li>Host: {destination.host || '-'} </li>
+        <li>Subset: {destination.subset || '-'} </li>
+        <li>Port: {destination.port ? destination.port.number : '-'} </li>
+      </ul>
+    );
+  }
+
+  renderTable(route: any, i: number) {
+    const resolvedColumns = resolve.columnChildren(this.columns());
+    const resolvedRows = resolve.resolve({
```

Comment:
I'm not familiar with the table-resolver library.
I assume this is just a migration from previous routerule table.

Reply:
yep, this is needed for migration. Table-resolver comes from reactabular.js the lib for building tables in PF. It is used in rourerule table too.